apps/web/src/components/Publication/Actions/Menu/CopyPostText.tsx

```diff
@@ -0,0 +1,47 @@
+import { Menu } from '@headlessui/react';
+import { ClipboardCopyIcon } from '@heroicons/react/outline';
+import type { Publication } from '@lenster/lens';
+import stopEventPropagation from '@lenster/lib/stopEventPropagation';
+import { Leafwatch } from '@lib/leafwatch';
+import { t, Trans } from '@lingui/macro';
+import clsx from 'clsx';
+import type { FC } from 'react';
+import toast from 'react-hot-toast';
+import { PUBLICATION } from 'src/tracking';
+
+interface CopyPostTextProps {
+  publication: Publication;
+}
+
+const CopyPostText: FC<CopyPostTextProps> = ({ publication }) => {
+  return (
+    <Menu.Item
+      as="div"
+      className={({ active }) =>
+        clsx(
+          { 'dropdown-active': active },
+          'm-2 block cursor-pointer rounded-lg px-4 py-1.5 text-sm'
+        )
+      }
+      onClick={async (event) => {
+        stopEventPropagation(event);
+        await navigator.clipboard.writeText(
+          publication?.metadata?.content || ''
+        );
+        toast.success(t`Copied to clipboard!`);
+        Leafwatch.track(PUBLICATION.COPY_TEXT, {
+          publication_id: publication.id
+        });
+      }}
+    >
+      <div className="flex items-center space-x-2">
+        <ClipboardCopyIcon className="h-4 w-4" />
+        <div>
+          <Trans>Copy post text</Trans>
```

Comment:
We can make the text dynamic like `Copy post text` and `Copt comment text` based on __typeName

Reply:
Done! I also added special logic for Mirror, (if it's a Mirror, it can say either "Copy post" or "Copy comment", depending on whether  the mirrored post is a post or a comment)